### PR TITLE
Mini-Toolbox Thumbnails

### PR DESCRIPTION
**Pull request by epeach**
Adds the framework that allows blocks to 'shadow' the value of other blocks. Tracks relationships between blocks and updates accordingly. Note: Currently only works for relationships between gamelab_clickedSpritePointer (the shadower) and gamelab_allSpritesWithAnimation (the shadow target). Future work would be to extend this feature to work with other types of blocks.
updatingBlock

Jira Ticket for Unit Testing: https://codedotorg.atlassian.net/browse/STAR-459

### CODE REVIEW

File: core/ui/block.js [hunk 1]
```diff
@@ -1500,11 +1500,31 @@ Blockly.Block.prototype.setParent = function(newParent) {
...
   }
 };
 
+Blockly.Block.prototype.shadowBlockValue_ = function() {
+  if(this.blockToShadow_){
+    let siblings = this.getRootBlock().childBlocks_;
+    siblings.forEach(function(sibling){
+      // If this block is the type of block that this block is to shadow, copy the preview value
+      if(this.blockToShadow_ === sibling.type){
+        // ToDo - Remove hard-coded values to indicate which input and title part to copy
```
[the remark this answers is not in the document]
Comment (epeach) [reply core/ui/block.js]: https://codedotorg.atlassian.net/browse/STAR-460

**Comment by epeach** [comment]
Jira Ticket for Unit Testing: https://codedotorg.atlassian.net/browse/STAR-459

File: core/ui/block.js [hunk 2]
```diff
@@ -1893,6 +1913,14 @@ Blockly.Block.prototype.shouldCopyOnDrag = function(){
   return this.copyOnDrag_ && !!parent && (parent.type === this.copyOnDrag_);
 };
 
+/**
+ * Sets the target block whose value this block should shadow
+ * @param target
+ */
+Blockly.Block.prototype.setBlockToShadow = function(target){
```
[the remark this answers is not in the document]
Comment (epeach) [reply core/ui/block.js]: There is a forthcoming PR in the cdo repo which will use this function. Expecting to have that up in the next few hours.

File: core/ui/block.js [hunk 3]
```diff
@@ -1500,11 +1504,53 @@ Blockly.Block.prototype.setParent = function(newParent) {
...
+      }
+    }.bind(this));
+  }
+};
+
+/**
+ * Tracks a field_image block to update with the value of this dropdown
+ * @param fieldImage - additional field to update
+ */
+Blockly.Block.prototype.addRelationalUpdate = function(fieldImage){
```
Comment (jmkulwik) [inline core/ui/block.js]: Would this work for any relational update, or just for a field_image block? The function name seems to imply the first, but the param name and comment seem to imply the second (same with the comment on getRelationalUpdateBlocks).

Comment (epeach) [reply core/ui/block.js]: Good catch! Originally, I wanted to make this more open ended and not just limited to field_image blocks, but in order to hit the deadline, I ended up pairing it down to just field_image. There is a ticket (https://codedotorg.atlassian.net/browse/STAR-460) for removing the parts of this that make it really only work for the field_image blocks. At that point, the comments should probably be updated to handle the broader applicability of this feature.

**Pull request merged**